python/Ganga/GPIDev/Credentials2/ICredentialRequirement.py

```diff
@@ -0,0 +1,57 @@
+from abc import abstractmethod
+
+from Ganga.GPIDev.Base.Objects import GangaObject
+from Ganga.GPIDev.Schema import Schema, Version
+
+from .ICredentialInfo import ICredentialInfo
+
+
+class ICredentialRequirement(GangaObject):
+    """
+    Requirements objects specify what is needed by a particular object in Ganga
+
+    All subclasses must specify ``info_class`` as well as all abstract methods
+    """
+
+    _schema = Schema(Version(1, 0))
+
+    _category = 'CredentialRequirement'
+    _name = 'ICredentialRequirement'
+    _hidden = 1
+
+    # This attribute refers to the ICredentialInfo subclass which can satisfy the requirements
+    info_class = None  # type: Type[ICredentialInfo]
+
+    def __init__(self, **kwargs):
+        super(ICredentialRequirement, self).__init__()
+        for key, value in kwargs.items():
+            if key not in self._schema.allItemNames():
+                raise Exception()
+            setattr(self, key, value)
```

Comment:
Wouldn't this be better in GangaObject as this is overloading the values which are constructed there?